fuse_core/include/fuse_core/graph.h

```diff
@@ -58,7 +58,7 @@ namespace fuse_core
  * be used in range-based for loops), an empty() method, and a front() method for directly accessing the first
  * member. When dereferenced, an iterator returns a const Constraint&.
  */
-using const_constraint_range = boost::any_range<const fuse_core::Constraint, boost::forward_traversal_tag>;
+using const_constraint_range = boost::any_range<const Constraint, boost::forward_traversal_tag>;
```

Comment:
I note that this header includes `transaction.h`, which also has this typedef, and in the same namespace. Note sure if you care. You probably don't want to assume anything about the typedefs in your includes.

Reply:
I noticed that as well. I definitely don't have a strong opinion, other than the current setup is inconsistent. Seems like I need to either:
(a) Move the typedefs in graph.h _inside_ the Graph class. This is what was done in the Transaction object, and it prevents any sort of name collision.
(b) Move all of these range definitions into a common file somewhere and include that whenever I need them.
Thoughts? Preferences?

Reply:
I like (a) better. Presumably, (b) would require that the `types.h` file (or whatever) includes `constraint.h`, `transaction.h`, etc. If that's true, then might we not get into weird circular dependencies where including the `types.h` file?